From 336ec784f53154fd04f632d0a3ecbadd05570aac Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 13 Mar 2018 09:34:58 +0900
Subject: [PATCH 4/4] Repaint on resize asynchronously

Otherwise nsViewManager::Refresh() is blocked infinitely when
CompositorOGL is enabled with EGL.

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 gfx/thebes/gfxPrefs.h | 2 +-
 widget/gtk/nsWindow.h | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/gfx/thebes/gfxPrefs.h b/gfx/thebes/gfxPrefs.h
index ccd7a22b938a..2d1d38128258 100644
--- a/gfx/thebes/gfxPrefs.h
+++ b/gfx/thebes/gfxPrefs.h
@@ -619,7 +619,7 @@ private:
   DECL_GFX_PREF(Live, "layers.progressive-paint",              ProgressivePaint, bool, false);
   DECL_GFX_PREF(Live, "layers.shared-buffer-provider.enabled", PersistentBufferProviderSharedEnabled, bool, false);
   DECL_GFX_PREF(Live, "layers.single-tile.enabled",            LayersSingleTileEnabled, bool, true);
-  DECL_GFX_PREF(Live, "layers.force-synchronous-resize",       LayersForceSynchronousResize, bool, true);
+  DECL_GFX_PREF(Live, "layers.force-synchronous-resize",       LayersForceSynchronousResize, bool, false);
 
   // We allow for configurable and rectangular tile size to avoid wasting memory on devices whose
   // screen size does not align nicely to the default tile size. Although layers can be any size,
diff --git a/widget/gtk/nsWindow.h b/widget/gtk/nsWindow.h
index 505aa018fe86..d88b7804ac6f 100644
--- a/widget/gtk/nsWindow.h
+++ b/widget/gtk/nsWindow.h
@@ -601,6 +601,9 @@ private:
 
     void CleanLayerManagerRecursive();
 
+    virtual bool SynchronouslyRepaintOnResize() override
+    { return false; }
+
     virtual int32_t RoundsWidgetCoordinatesTo() override;
 
     /**
-- 
2.14.1

